clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1246,7 +1246,7 @@ public void assign(Collection<TopicPartition> partitions) {
                     // NOTE: since the consumed position has already been updated, we must not allow
                     // wakeups or any other errors to be triggered prior to returning the fetched records.
                     if (fetcher.sendFetches() > 0 || client.hasPendingRequests()) {
```

Comment:
Would it be more robust to wrap the whole if statement in a try/catch?

Reply:
if we `try-catch` the block, we still need to "remember" whatever errors was thrown so that later callers can still be informed of such errors otherwise they will be swallowed silently. So I think "not checking them" is actually better since this is a one-off best-effort send shot anyways.

Reply:
The errors that we are trying to handle here are already remembered anyway (primarily inside of `Metadata`). So they will still get thrown on the next call to `poll`.